Cython/Shadow.pyi

```diff
@@ -1,44 +1,229 @@
+
+# Type checkers assume typing_extensions is always present
+from typing_extensions import Literal, ParamSpec, overload, final as final
+
+# Normally all imports aren't exported in stub files but we can explicitly expose
+# imports by using import ... as ... (with the same name) which was done for
+# dataclasses and the final decorator.
+
+__version__: str
+
+# Predefined types
 
 int = py_int
 long = py_int
 longlong = py_int
 short = py_int
 char = py_int
+
 sint = py_int
 slong = py_int
 slonglong = py_int
 sshort = py_int
 schar = py_int
+
 uint = py_int
 ulong = py_int
 ulonglong = py_int
 ushort = py_int
 uchar = py_int
+
 size_t = py_int
 Py_ssize_t = py_int
-Py_UCS4 = Union[py_int, str]
-Py_UNICODE = Union[py_int, str]
+
+Py_UCS4 = py_int | str
+Py_UNICODE = py_int | str
+
 float = py_float
 double = py_float
 longdouble = py_float
+
 complex = py_complex
 floatcomplex = py_complex
 doublecomplex = py_complex
 longdoublecomplex = py_complex
+
 bint = py_bool
-void = Union[None]
+void = Type[None]
 basestring = py_str
 unicode = py_str
 
-gs: Dict[str, Any]  # Should match the return type of globals()
+gs: dict[str, Any]  # Should match the return type of globals()
+compiled: bool
+
 
 _T = TypeVar('_T')
+_P = ParamSpec('_P')
+_C = TypeVar('_C', bound='Callable')
+_TypeT = TypeVar('_TypeT', bound='Type')
+_Decorator = Callable[[_C], _C]
+
+
+_func_deco: _Decorator
+
+cfunc = ccall = compile = _func_deco
+
+def locals(**kwargs: Any) -> _Decorator: ...
+
+def _class_deco(__cls: _TypeT) -> _TypeT: ...
+
+cclass = internal = c_api_binop_methods = type_version_tag = no_gc_clear = no_gc = _class_deco
+
+# May be a bit hard to read but essentially means:
+# > Returns a callable that takes another callable with these parameters and *some*
+# > return value, then returns another callable with the same parameters but
+# > the return type is the previous 'type' parameter.
+# On Python 3.5, the latest version of Mypy available is 0.910 which doesn't understand ParamSpec
+def returns(__type: Type[_T]) -> Callable[[Callable[_P, object]], Callable[_P, _T]]: ...  # type: ignore
+
+def exceptval(__val: Any, *, check: bool = False) -> _Decorator: ...
+
+class _EmptyDecoratorAndManager(object):
+    @overload
+    def __call__(self, __val: bool) -> _Decorator: ...
+
+    @overload
+    def __call__(self, __func: _C) -> _C: ...
+
+    def __enter__(self) -> None: ...
+
+    def __exit__(
+        self,
+        exc_type: Optional[Type[BaseException]],
+        exc: Optional[BaseException],
+        tb: Optional[TracebackType]
+    ) -> None: ...
+_empty_decorator_and_manager: _EmptyDecoratorAndManager
+
+@overload
+def _compiler_directive(__func: _C) -> _C: ...
+
+@overload
+def _compiler_directive(__val: bool = ...) -> _Decorator: ...
+
+# These all come from 'Compiler directives' on Source Files and Compilation.
+# The following directives are missing as they need to be global:
+# - annotation_typing
```

Comment:
This is no longer true for `annotation_typing`. My inclination is to merge this even if slightly incomplete though